src/Workspaces/Core/Portable/Storage/SQLite/v2/SQLitePersistentStorage.Accessor.cs

```diff
@@ -26,10 +24,13 @@ internal partial class SQLitePersistentStorage
         /// logic around cancellation/pooling/error-handling/etc, while still hitting different
         /// db tables.
         /// </summary>
-        private abstract class Accessor<TKey, TWriteQueueKey, TDatabaseId>
+        private abstract class Accessor<TKey, TWriteQueueKey, TDatabaseId> 
+            where TDatabaseId : struct
         {
             protected readonly SQLitePersistentStorage Storage;
 
+            private readonly ImmutableArray<(string name, string type)> _primaryKeys;
```

Comment:
by having the accessor know the names/types of its primary-key columns we can unify a bunch of code that was previously doing this manually.